File: golang/lib/testsuite/test_configuration.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package testsuite

import "github.com/kurtosis-tech/kurtosis-client/golang/lib/services"

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type TestConfiguration struct {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	SetupTimeoutSeconds uint32

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	RunTimeoutSeconds uint32

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	IsPartitioningEnabled bool

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	FilesArtifactUrls map[services.FilesArtifactID]string

}

File: rust/lib/src/testsuite/test_configuration.rs
use std::collections::HashMap;

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
pub struct TestConfiguration {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub test_setup_timeout_seconds: u32,

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub test_run_timeout_seconds: u32,

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub is_partitioning_enabled: bool,

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
    pub files_artifact_urls: HashMap<String, String>,
}